pkg/globalnet/controllers/ipam/gatewaymonitor_test.go

```diff
@@ -104,74 +111,128 @@ var _ = Describe("Service monitoring", func() {
-				ContainSubstring(t.awaitServiceGlobalIP(service.Name, localCIDR)))
+				t.createService(service)
+				globalIP := t.awaitServiceGlobalIP(service.Name)
+				t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))
+			})
 		})
-	})
 
-	When("a Service's global IP is updated", func() {
-		BeforeEach(func() {
-			service.Annotations[ipam.SubmarinerIpamGlobalIp] = "169.254.1.10"
+		When("the Service is subsequently exported", func() {
+			It("should eventually assign a global IP and add an appropriate IP tables rule", func() {
+				service = t.createService(service)
+				t.awaitNoServiceGlobalIP(service)
+				t.ipt.AwaitNoRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIPSubstr))
+
+				t.createSvcExport(service)
+				globalIP := t.awaitServiceGlobalIP(service.Name)
+				t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))
+			})
 		})
 
-		It("should update the appropriate IP tables rule", func() {
-			t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain,
-				ContainSubstring(service.Annotations[ipam.SubmarinerIpamGlobalIp]))
+		When("a Service Export is created before service", func() {
+			It("should assign a global IP and add an appropriate IP tables rule after service creation", func() {
+				t.createSvcExport(service)
+				t.ipt.AwaitNoRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIPSubstr))
 
-			service.Annotations[ipam.SubmarinerIpamGlobalIp] = "169.254.1.20"
-			_, err := t.k8sClient.CoreV1().Services(namespace).Update(service)
-			Expect(err).To(Succeed())
-			t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain,
-				ContainSubstring(service.Annotations[ipam.SubmarinerIpamGlobalIp]))
+				t.createService(service)
+				globalIP := t.awaitServiceGlobalIP(service.Name)
+				t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))
+			})
 		})
-	})
 
-	When("a Service with a global IP is removed", func() {
-		It("should remove the appropriate IP tables rule", func() {
-			globalIP := t.awaitServiceGlobalIP(service.Name, localCIDR)
-			t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))
+		When("the Service Export is deleted", func() {
+			It("should eventually remove global IP and the IP tables rule", func() {
+				service = t.createService(service)
+				t.createSvcExport(service)
+				globalIP := t.awaitServiceGlobalIP(service.Name)
+				t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))
 
-			Expect(t.k8sClient.CoreV1().Services(namespace).Delete(service.Name, nil)).To(Succeed())
-			t.ipt.AwaitNoRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))
+				Expect(t.dynClient.Resource(*t.svcExGvr).Namespace(service.Namespace).Delete(service.Name, nil)).To(Succeed())
+				t.ipt.AwaitNoRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIPSubstr))
+			})
```

Comment:
I suggested to put this in the next parent spec below after L171 since it relies on the service already assigned, like the update case. Saves some redundant code that is done in `JustBeforeEach` on L167:

```
    When("a Service with a global IP is unexported", func() {
        It("should unassign its global IP and remove the appropriate IP tables rule", func() {
            globalIP := t.awaitServiceGlobalIP(service.Name, localCIDR) 
            t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))

            Expect(t.dynClient.Resource(*t.svcExGvr).Namespace(service.Namespace).Delete(service.Name, nil)).To(Succeed())
	    t.ipt.AwaitNoRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))
            t.awaitNoServiceGlobalIP(service)
        })
    })


```

Reply:
Done.